data_pipeline/surveymonkey/surveymonkey_etl.py

```diff
@@ -30,13 +30,71 @@ def get_survey_list(access_token: str) -> list:
 def get_survey_question_details(access_token: str, survey_id: str) -> list:
     headers = get_surveymonkey_api_headers(access_token)
     response = requests.get(
-        f'https://api.surveymonkey.com/v3/surveys/{survey_id}/details',
+        f"https://api.surveymonkey.com/v3/surveys/{survey_id}/details",
         headers=headers
     )
     reponse_json = response.json()
     return reponse_json
 
 
+def get_no_answer_options_for_question_json() -> dict:
+    return {
+        "id": None,
+        "text": None,
+        "type": None,
+        "weight": None
+    }
+
+
+def parse_answer_options_in_question_answer_json(
+    question_answer_response_json: dict,
+    option_key: str
+) -> dict:
+    if isinstance(question_answer_response_json[option_key], list):
+        result_list = []
+        for answer_option in question_answer_response_json[option_key]:
+            result_list.append({
+                "id": answer_option.get("id"),
+                "text": answer_option.get("text"),
+                "type": answer_option.get("type"),
+                "weight": answer_option.get("weight")
+            })
+        return result_list
+    return [{
+        "id": question_answer_response_json[option_key].get("id"),
+        "text": question_answer_response_json[option_key].get("text"),
+        "type": question_answer_response_json[option_key].get("type"),
+        "weight": question_answer_response_json[option_key].get("weight")
+    }]
+
+
+def parse_answers_in_question_json(question_response_json: dict) -> dict:
+    result = {
+        "choices": [get_no_answer_options_for_question_json()],
+        "other": [get_no_answer_options_for_question_json()],
+        "rows": [get_no_answer_options_for_question_json()],
+        "cols": [get_no_answer_options_for_question_json()]
+    }
+    if "answers" in question_response_json:
+        if "choices" in question_response_json["answers"]:
```

Comment:
I don't know much about the survey response, but if you can use `parse_answer_options_in_question_answer_json` for all of the answers, it seems like here you could iterate over a list of keys to reduce duplication?
e.g.

```python
for key in ["choices", "other", "rows", "cols"]:
    result[key] = parse_answer_options_in_question_answer_json(
        question_response_json["answers"],
        key
    )
```

Reply:
Added:

```python
    for key in ["choices", "other", "rows", "cols"]:
            if key in question_response_json["answers"]:
                result[key] = parse_answer_options_in_question_answer_json(
                    question_response_json["answers"][key]
                )
```